### PR TITLE
feat(schema validation): check references schema names and types before changing config

### PR DESCRIPTION
**Pull request by thalesmg**
Fixes https://emqx.atlassian.net/browse/EMQX-12368

Release version: e5.7.1

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_schema_registry/src/emqx_schema_registry.erl [hunk 1]
```diff
@@ -64,6 +67,14 @@ get_serde(SchemaName) ->
             {ok, Serde}
     end.
 
+-spec does_type_exist(schema_name()) -> boolean().
+does_type_exist(SchemaName) ->
```
Comment (zmstone) [inline apps/emqx_schema_registry/src/emqx_schema_registry.erl:71]: the convention is to use is_ prefix for bool check result functions.
maybe is_existing_type

**Review by zmstone — COMMENTED** [review]
`emqx_schema_validation` app should depend on `emqx_schema_registry`

File: apps/emqx_schema_registry/src/emqx_schema_registry.erl (hunk 1 above)
Comment (thalesmg) [reply apps/emqx_schema_registry/src/emqx_schema_registry.erl]: :heavy_check_mark:

**Comment by thalesmg** [comment]
> `emqx_schema_validation` app should depend on `emqx_schema_registry`

:heavy_check_mark:

**Review by zmstone — DISMISSED** [review]

**Review by zmstone — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-2602)

**Pull request merged**